### PR TITLE
Add convert_text_to_mds to CLI

### PR DESCRIPTION
**Pull request by KuuCi**
This PR allows users to call llmfoundry convert_text_to_mds {ARGS} while maintaining correctness with existing convert_text_to_mds script. The motivation is for DLE where we want to make the CLI much more intuitive in the docker images

### CODE REVIEW

**Comment by KuuCi** [comment]
test-data-txt-LJgtNP runs:
`llmfoundry convert_text_to_mds \
    --output_folder my-copy-shakespeare \
    --input_folder shakespeare \
    --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b \
    --compression zstd`
    
test-data-txt-du60oE runs:
`python data_prep/convert_text_to_mds.py \
    --output_folder my-copy-shakespeare \
    --input_folder shakespeare \
    --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b \
    --compression zstd
image: mosaicml/llm-foundry:2.3.1_cu121-latest`

**Comment by KuuCi** [comment]
txt-cli-GVwCPT Runs:
`  llmfoundry convert_text_to_mds \
    --output-folder my-copy-shakespeare \
    --input-folder shakespeare \
    --concat-tokens 2048 --tokenizer EleutherAI/gpt-neox-20b \
    --compression zstd`
    
txt-orig-Zagm3I Runs:
`python convert_text_to_mds.py \
    --output_folder my-copy-shakespeare \
    --input_folder shakespeare \
    --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b \
    --compression zstd`

**Comment by KuuCi** [comment]
txt-cli-Y2HXxb runs:
`  llmfoundry data_prep convert_text_to_mds \
    --output-folder my-copy-shakespeare \
    --input-folder shakespeare \
    --concat-tokens 2048 --tokenizer EleutherAI/gpt-neox-20b \
    --compression zstd`
    
txt-orig-tFXNED runs:
`  python convert_text_to_mds.py \
    --output_folder my-copy-shakespeare \
    --input_folder shakespeare \
    --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b \
    --compression zstd`
    
image

File: llmfoundry/command_utils/__init__.py [hunk 1]
```diff
@@ -26,4 +37,13 @@
     'eval_from_yaml',
     'convert_dataset_hf',
     'convert_dataset_hf_from_args',
+    'convert_text_to_mds',
+    'convert_text_to_mds_from_args',
+    'maybe_create_object_store_from_uri',
```
Comment (dakinggg) [inline llmfoundry/command_utils/__init__.py:42]: this is from composer and shouldn't be reexported, same with `parse_uri`

File: llmfoundry/command_utils/__init__.py [hunk 2]
```diff
@@ -26,4 +37,13 @@
...
+    'parse_uri',
+    'download_and_convert',
```
Comment (dakinggg) [inline llmfoundry/command_utils/__init__.py:44]: I don't think all these helpers should be exported as part of the command_utils module

File: llmfoundry/command_utils/data_prep/convert_text_to_mds.py [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+            trust_remote_code=trust_remote_code,
+        )
+        eos_text = built_tokenizer.eos_token
+
+    # now that we have validated them, change BOS/EOS to strings
+    if bos_text is None:
+        bos_text = ''
+    if eos_text is None:
+        eos_text = ''
+    _configure_logging(logging_level)
```
Comment (dakinggg) [inline llmfoundry/command_utils/data_prep/convert_text_to_mds.py:553]: this should probably happen at the start of this function?

File: scripts/data_prep/convert_text_to_mds.py [hunk 4]
```diff
@@ -211,454 +113,23 @@ def parse_args() -> Namespace:
...
-    _configure_logging(args.logging_level)
-    convert_text_to_mds(
-        tokenizer_name=args.tokenizer,
+    convert_text_to_mds_from_args(
         output_folder=args.output_folder,
         input_folder=args.input_folder,
+        compression=args.compression,
         concat_tokens=args.concat_tokens,
-        eos_text=args.eos_text,
+        tokenizer=args.tokenizer,
```
Comment (dakinggg) [inline scripts/data_prep/convert_text_to_mds.py:126]: why did you change this from tokenizer_name to tokenizer?

**Review by dakinggg — APPROVED** [review]
